docs/guide/http_cache.md

```diff
@@ -730,11 +737,167 @@ set req.http.X-User-Hash = "b1731d46b0e7a375a5b024e950fdb8d49dd25af85a5c7dd5116a
+```
+// Sub-routine to get client user hash, for context-aware HTTP cache.
+sub ez_user_hash {
+
+    // Prevent tampering attacks on the hash mechanism
+    if (req.restarts == 0
+        && (req.http.accept ~ "application/vnd.fos.user-context-hash"
+            || req.http.x-user-hash
+        )
+    ) {
+        return (synth(400));
+    }
+
+    if (req.restarts == 0 && (req.method == "GET" || req.method == "HEAD")) {
+        // Get User (Context) hash, for varying cache by what user has access to.
+        // https://doc.ez.no/display/EZP/Context+aware+HTTP+cache
+
+        // Request to eng siteaccess, so we should use other X-User-Hash for it
+        if (req.url ~ "^/eng") {
+            // Anonymous user w/o session => Use hardcoded anonymous hash to avoid backend lookup for hash
+            if (req.http.Cookie !~ "eZSESSID_eng" && !req.http.authorization) {
+                // Note: You should then update it every time anonymous user rights change.
+                set req.http.X-User-Hash = "baf9acf7ca78e370eac69f87f27e4ab8e674ced83750b4189e216cc05d2eb301";
+            }
+            // Pre-authenticate request to get shared cache, even when authenticated
+            else {
+                set req.http.x-fos-original-url    = req.url;
+                set req.http.x-fos-original-accept = req.http.accept;
+                set req.http.x-fos-original-cookie = req.http.cookie;
+                // Clean up cookie for the hash request to only keep session cookie, as hash cache will vary on cookie.
+                set req.http.cookie = ";" + req.http.cookie;
+                set req.http.cookie = regsuball(req.http.cookie, "; +", ";");
+                set req.http.cookie = regsuball(req.http.cookie, ";(eZSESSID_eng[^=]*)=", "; \1=");
+                set req.http.cookie = regsuball(req.http.cookie, ";[^ ][^;]*", "");
+                set req.http.cookie = regsuball(req.http.cookie, "^[; ]+|[; ]+$", "");
+                set req.http.accept = "application/vnd.fos.user-context-hash";
+                set req.url = "/_fos_user_context_hash";
+    
+                // Force the lookup, the backend must tell how to cache/vary response containing the user hash
+                return (hash);
+            }
+        }
+        
+        // Request to nor siteaccess, so we should use other X-User-Hash for it
+        if (req.url ~ "^/nor") {
+            // Anonymous user w/o session => Use hardcoded anonymous hash to avoid backend lookup for hash
+            if (req.http.Cookie !~ "eZSESSID_nor" && !req.http.authorization) {
+                // Note: You should then update it every time anonymous user rights change.
+                set req.http.X-User-Hash = "a33ba7050ec3b848b266ef187623417b88b9df4b90483b7ef6582aa54ee72ee7";
+            }
+            // Pre-authenticate request to get shared cache, even when authenticated
+            else {
+                set req.http.x-fos-original-url    = req.url;
```

Comment:
Afaik we don't need to duplicate this section, the VCL session cookie pattern should be able to handle `_whatever` on session name.

Also lets move this syntax to be for ezplatform-http-cache VCL syntax, so it's the same across 1.7(.7) / 1.13 / 2.0 versions. In this xkey VCL flavor session cookie logic is already handled in `vcl_recv()` so less complicated logic in `ez_user_context_hash ()` to explain how to modify.

Optional: Move the VCL part of the doc to 1.13 to avoid having to explain which VCL type example is for, most users are moving to 1.13 / 2.0 anyway.